### PR TITLE
set up generic repository

### PR DESCRIPTION
**Pull request by Mmishaaa**

### CODE REVIEW

**Review by VadimZharikov — APPROVED** [review]

File: Tinder.DAL/Repositories/GenericRepository.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public class GenericRepository<TEntity> : IGenericRepository<TEntity> where TEntity : BaseEntity
+    {
+        protected ApplicationDbContext _context;
+        public GenericRepository(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+        public virtual async Task<TEntity> CreateAsync(TEntity entity, CancellationToken cancellationToken)
+        {
+           await _context.Set<TEntity>().AddAsync(entity, cancellationToken);
```
Comment (pavelGagarin0870) [inline Tinder.DAL/Repositories/GenericRepository.cs]: You can move _context.Set<TEntity>() to property, initialize on ctor and re-use it

File: Tinder.DAL/Repositories/GenericRepository.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+           await _context.SaveChangesAsync(cancellationToken);
+           return entity;
+        }
+
+        public virtual async Task<IEnumerable<TEntity>> GetAllAsync(CancellationToken cancellationToken)
```
Comment (pavelGagarin0870) [inline Tinder.DAL/Repositories/GenericRepository.cs]: async/await can be removed. Also can be transformed to the expression method

File: Tinder.DAL/Repositories/GenericRepository.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            var entities = await _context.Set<TEntity>()
+                .AsNoTracking()
+                .ToListAsync(cancellationToken);
+            return entities;
+        }
+
+        public virtual async Task<TEntity?> GetByIdAsync(Guid id, CancellationToken cancellationToken)
+        {
+            var entity = await _context.Set<TEntity>()
```
Comment (pavelGagarin0870) [inline Tinder.DAL/Repositories/GenericRepository.cs]: async/await can be removed. Also can be transformed to the expression method

File: Tinder.DAL/Repositories/GenericRepository.cs (hunk 2 above)
Comment (pavelGagarin0870) [reply Tinder.DAL/Repositories/GenericRepository.cs]: public virtual Task<IEnumerable<TEntity>> GetAllAsync(CancellationToken cancellationToken) => 
         _context.Set<TEntity>()
                .AsNoTracking()
                .ToListAsync(cancellationToken)

**Review by pavelGagarin0870 — CHANGES_REQUESTED** [review]

File: Tinder.DAL/Repositories/GenericRepository.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+{
+    public class GenericRepository<TEntity> : IGenericRepository<TEntity> where TEntity : BaseEntity
+    {
+        protected ApplicationDbContext _context;
+        protected DbSet<TEntity> _dbSet; 
+        public GenericRepository(ApplicationDbContext context)
+        {
+            _context = context;
+            _dbSet = _context.Set<TEntity>();
+        }
```
Comment (moltenessence) [inline Tinder.DAL/Repositories/GenericRepository.cs:16]: Separate constructor/methods with empty lines

File: Tinder.DAL/Repositories/GenericRepository.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
+﻿using Tinder.DAL.Interfaces;
+using Tinder.DAL.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Tinder.DAL.Repositories
+{
+    public class GenericRepository<TEntity> : IGenericRepository<TEntity> where TEntity : BaseEntity
+    {
+        protected ApplicationDbContext _context;
+        protected DbSet<TEntity> _dbSet;
```
Comment (moltenessence) [inline Tinder.DAL/Repositories/GenericRepository.cs:10]: please add empty line before constructor definition

**Review by moltenessence — APPROVED** [review]

**Review by pavelGagarin0870 — APPROVED** [review]

**Pull request merged**